pkg/internal/limitrange/transformer_test.go

```diff
@@ -805,6 +789,188 @@ func TestTransformerMultipleContainer(t *testing.T) {
 				},
 			}},
 		},
+	}, {
+		description: "limitRange with default requests and no resources on 2 step containers and a sidecar",
+		limitranges: []corev1.LimitRangeItem{{
+			Type: corev1.LimitTypeContainer,
+			Default: corev1.ResourceList{
+				corev1.ResourceCPU:              resource.MustParse("2"),
+				corev1.ResourceMemory:           resource.MustParse("200Mi"),
+				corev1.ResourceEphemeralStorage: resource.MustParse("2Gi"),
+			},
+			DefaultRequest: corev1.ResourceList{
+				corev1.ResourceCPU:              resource.MustParse("1"),
+				corev1.ResourceMemory:           resource.MustParse("100Mi"),
+				corev1.ResourceEphemeralStorage: resource.MustParse("1Gi"),
+			},
+		}},
+		podspec: corev1.PodSpec{
+			InitContainers: []corev1.Container{{
+				Name:  "bar",
+				Image: "foo",
+			}},
+			Containers: []corev1.Container{{
+				Name:  "step-foo",
+				Image: "baz",
+			}, {
+				Name:  "step-bar",
+				Image: "baz",
+			}, {
+				Name:  "sidecar-fizz",
+				Image: "baz",
+			}},
+		},
+		want: corev1.PodSpec{
```

Comment:
it's hard to tell with this test case which containers are receiving the divided requests vs the full requests. we might need to update the test function to compare the container names as well as the resources, so the test can ensure that the correct containers are getting the correct resources.

Reply:
I've updated the sidecar tests and added support in cmpRequestsAndLimits to also check the Container Name matches but only if present in "wants"